src/tasks/withdraw/traded_tokens.ts

```diff
@@ -35,13 +48,35 @@ export async function getAllTradedTokens(
     });
     console.log(`Processed events from block ${fromBlock} to ${toBlock}`);
   } catch (error) {
-    // Infura throws "too many events" error. Other nodes time out.
-    if (!(isErrorTooManyEvents(error) || isTimeout(error))) {
-      throw error;
+    switch (decodeError(error)) {
+      // If the query is too large, Infura throws "too many events" error while
+      // other nodes time out.
+      case ProviderError.Timeout:
+      case ProviderError.TooManyEvents:
+        console.log(
+          `Failed to process events from block ${fromBlock} to ${toBlock}, reducing range...`,
+        );
+        break;
+      case ProviderError.BlockNotFound:
+        if (fromBlock < toBlock) {
+          console.log(
+            `Block not found when retrieving blocks ${fromBlock} to ${toBlock}, skipping last block...`,
+          );
+          return await getAllTradedTokens(
+            settlement,
+            fromBlock,
+            toBlock - 1,
```

Comment:
maybe use `"latest"` here?

Reply:
Done!

Reply:
Since there were extra changes needed, please double check that everything still looks ok. :slightly_smiling_face: